src/System.Net.Security/src/System/Net/_SSPIWrapper.cs

```diff
@@ -0,0 +1,873 @@
+                case Interop.SecurityStatus.MessageAltered:
+                    return "Message altered";
+                case Interop.SecurityStatus.UntrustedRoot:
+                    return "Untrusted root";
+                default:
+                    return "0x" + errorCode.ToString("x", NumberFormatInfo.InvariantInfo);
+            }
+        }
+    } // class SSPIWrapper
+
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal class StreamSizes
+    {
+        public int header;
+        public int trailer;
+        public int maximumMessage;
+        public int buffersCount;
+        public int blockSize;
+
+        internal unsafe StreamSizes(byte[] memory)
+        {
+            fixed (void* voidPtr = memory)
+            {
+                IntPtr unmanagedAddress = new IntPtr(voidPtr);
+                try
+                {
+                    header = (int)checked((uint)Marshal.ReadInt32(unmanagedAddress));
+                    trailer = (int)checked((uint)Marshal.ReadInt32(unmanagedAddress, 4));
+                    maximumMessage = (int)checked((uint)Marshal.ReadInt32(unmanagedAddress, 8));
+                    buffersCount = (int)checked((uint)Marshal.ReadInt32(unmanagedAddress, 12));
+                    blockSize = (int)checked((uint)Marshal.ReadInt32(unmanagedAddress, 16));
+                }
+                catch (OverflowException)
+                {
+                    GlobalLog.Assert(false, "StreamSizes::.ctor", "Negative size.");
+                    throw;
+                }
+            }
+        }
+        public static readonly int SizeOf = Marshal.SizeOf<StreamSizes>();
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal class SecSizes
+    {
+        public readonly int MaxToken;
+        public readonly int MaxSignature;
+        public readonly int BlockSize;
+        public readonly int SecurityTrailer;
+
+        internal unsafe SecSizes(byte[] memory)
+        {
+            fixed (void* voidPtr = memory)
+            {
+                IntPtr unmanagedAddress = new IntPtr(voidPtr);
+                try
+                {
+                    MaxToken = (int)checked((uint)Marshal.ReadInt32(unmanagedAddress));
+                    MaxSignature = (int)checked((uint)Marshal.ReadInt32(unmanagedAddress, 4));
+                    BlockSize = (int)checked((uint)Marshal.ReadInt32(unmanagedAddress, 8));
+                    SecurityTrailer = (int)checked((uint)Marshal.ReadInt32(unmanagedAddress, 12));
+                }
+                catch (OverflowException)
+                {
+                    GlobalLog.Assert(false, "SecSizes::.ctor", "Negative size.");
+                    throw;
+                }
+            }
+        }
+        public static readonly int SizeOf = Marshal.SizeOf<SecSizes>();
+    }
+
+    // TODO (Issue #3114): Move to Interop. 
+    // From Schannel.h
+    [StructLayout(LayoutKind.Sequential)]
+    internal class SslConnectionInfo
```

Comment:
It's somewhat unusual to see `class` decorated with a StructLayout attribute; did you mean `struct`?